agent/agentserver/server_test.go

```diff
@@ -276,13 +277,113 @@ func TestReadinessCheckHandler(t *testing.T) {
 	}
 }
 
+func TestReadinessCheckHandlerCache(t *testing.T) {
+	testErr := errors.New("test-err")
+	type call struct {
+		success bool
+	}
+
+	for _, tc := range []struct {
+		desc              string
+		readinessCacheTTL time.Duration
+		waitInvalidation  bool
+		expCalls          []call
+		setupMocks        func(m *serverMocks)
+	}{
+		{
+			desc:              "call 1 succeeds, so second call succeeds without checks",
+			readinessCacheTTL: 10 * time.Minute,
+			waitInvalidation:  false,
+			expCalls:          []call{{success: true}, {success: true}},
+			setupMocks: func(m *serverMocks) {
+				m.sched.EXPECT().Probe().Return(nil).Times(1)
```

Comment:
`m.sched.EXPECT().Probe().Return(nil)`

`Times(1)` is the default and is usually omitted

Reply:
Done.